### PR TITLE
[HDInsight] Add two parameters `--resource-provider-connection` and `--enable-private-link` to support relay outbound and private link

### PR DESCRIPTION
**Pull request by aim-for-better**
**Description**  

Add two parameters `--resource-provider-connection` and `--private-link` to support relay outbound and private link feature.

**Testing Guide**  

**History Notes**  

[HDInsight]: az hdinsight create: Add two parameters `--resource-provider-connection` and `--enable-private-link` to support relay outbound and private link feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
HDInsight

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 1]
```diff
@@ -217,6 +218,13 @@ def load_arguments(self, _):
                    arg_group='Autoscale Configuration',
                    help='The scheduled workernode count.')
 
+        # relay outbound and private link
+        c.argument('resource_provider_connection', options_list=['--resource-provider-connection', '--rp-connection'],
+                   arg_group='Relay Outbound And PrivateLink',
+                   arg_type=get_enum_type(ResourceProviderConnection), help='The resource provider connection type')
+        c.argument('private_link', arg_group='Relay Outbound And PrivateLink',
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:225]: I don't quite understand this design. From the perspective of the argument name, it looks like Relay bound is a subproperty of PrivateLink. But from the perspective of the argument group name, they are equipotent resources.

Comment (aim-for-better) [reply src/azure-cli/azure/cli/command_modules/hdinsight/_params.py]: Hi @haroldrandom I have updated the PR could you please review it again? Thanks

File: src/azure-cli/azure/cli/command_modules/hdinsight/_params.py [hunk 2]
```diff
@@ -217,6 +217,13 @@ def load_arguments(self, _):
                    arg_group='Autoscale Configuration',
                    help='The scheduled workernode count.')
 
+        # relay outbound and private link
+        c.argument('resource_provider_connection', options_list=['--resource-provider-connection', '--rp-connection'],
+                   arg_group='Resource provider connection',
+                   arg_type=get_enum_type(ResourceProviderConnection), help='The resource provider connection type')
+        c.argument('enable_private_link', arg_group='Private Link', arg_type=get_three_state_flag(),
+                   help='Indicates whether enable the private link or not.')
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/hdinsight/_params.py:225]:
```suggestion
                   help='Indicate whether enable the private link or not.')
```

**Review by haroldrandom — APPROVED** [review]

**Pull request merged**